ambry-clustermap/src/main/java/com/github/ambry/clustermap/CloudServiceClusterChangeHandler.java

```diff
@@ -95,9 +95,12 @@ public void registerClusterMapListener(ClusterMapChangeListener clusterMapChange
   }
 
   @Override
-  public void getReplicaIdsByStates(Map<ReplicaState, List<AmbryReplica>> replicasByState,
-      Set<ReplicaState> states, AmbryPartition partition) {
-    throw new UnsupportedOperationException("Get snapshot of replica state is temporarily not supported");
+  public void getReplicaIdsByStates(Map<ReplicaState, List<AmbryReplica>> replicasByState, Set<ReplicaState> states,
```

Comment:
Should we honor `states` like in HelixClusterManager?

Reply:
Made a change to add CloudServiceReplica to LEADER list or STANDBY list only. Please see the code.